modules/programs/atuin.nix

```diff
@@ -211,6 +211,12 @@ in {
         };
       })
       (mkIf isDarwin {
+        programs.atuin.settings = {
+          daemon = {
+            socket_path = lib.mkDefault "${config.xdg.runtimeHome}/atuin.sock";
```

Comment:
Looking at the Atuin source code it seems it uses the data directory when the runtime directory is unset, so for MacOS, perhaps we could use that path instead? Unfortunately I don't know what path that is for Macs, though. Perhaps @water-sucks knows?

See https://github.com/atuinsh/atuin/blob/0056c72d89c1a01122ffdaebf15e0df6b63c8bf2/crates/atuin-common/src/utils.rs#L93-L95